src/codec/mod.rs

```diff
@@ -385,34 +385,17 @@ impl TryFrom<Bytes> for ExceptionResponse {
             ));
         }
         let function = fn_err_code - 0x80;
-        let exception = Exception::try_from(rdr.read_u8()?)?;
+        let exception = Exception::from(rdr.read_u8()?);
         Ok(ExceptionResponse {
             function: FunctionCode::new(function),
             exception,
         })
     }
 }
 
-impl TryFrom<u8> for Exception {
-    type Error = Error;
-
-    fn try_from(code: u8) -> Result<Self, Self::Error> {
-        use crate::frame::Exception::*;
-        let ex = match code {
-            0x01 => IllegalFunction,
-            0x02 => IllegalDataAddress,
-            0x03 => IllegalDataValue,
-            0x04 => ServerDeviceFailure,
-            0x05 => Acknowledge,
-            0x06 => ServerDeviceBusy,
-            0x08 => MemoryParityError,
-            0x0A => GatewayPathUnavailable,
-            0x0B => GatewayTargetDevice,
-            _ => {
-                return Err(Error::new(ErrorKind::InvalidData, "Invalid exception code"));
-            }
-        };
-        Ok(ex)
+impl From<u8> for Exception {
```

Comment:
I suggest to delete this `impl From`.

As a user of this crate you probably do not want to convert arbitrary `u8` values implicitly into a Modbus exception code with `.into()`. With `Exception::new()` we now have an explicit constructor to make such a conversion explicit.

I generally try to avoid `From`/`TryFrom` implementations between types of different abstraction levels. These should be explicitly converted with properly named methods.

Please then also update the changelog accordingly.